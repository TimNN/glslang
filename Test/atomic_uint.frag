#version 420 core

layout(binding = 0) uniform atomic_uint counter;

uint func(atomic_uint c)
{
    return atomicCounterIncrement(c);
}

uint func2(out atomic_uint c) // ERROR
{
    return counter;
}

void main()
{
     atomic_uint non_uniform_counter; // ERROR
     uint val = atomicCounter(counter);
     atomicCounterDecrement(counter);
}
