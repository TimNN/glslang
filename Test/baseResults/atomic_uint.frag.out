atomic_uint.frag
Warning, version 420 is not yet complete; most version-specific features are present, but some are missing.
ERROR: 0:10: 'atomic_uint' : samplers and atomic_uints cannot be output parameters 
ERROR: 0:12: 'return' : type does not match, or is not convertible to, the function's return type 
ERROR: 0:18: 'atomic_uint' : atomic_uints can only be used in uniform variables or function parameters: non_uniform_counter
ERROR: 3 compilation errors.  No code generated.


Shader version: 420
ERROR: node is still EOpNull!
0:5  Function Definition: func(au1; (uint)
0:5    Function Parameters: 
0:5      'c' (in atomic_uint)
0:7    Sequence
0:7      Branch: Return with expression
0:7        Function Call: atomicCounterIncrement(au1; (uint)
0:7          'c' (in atomic_uint)
0:10  Function Definition: func2(au1; (uint)
0:10    Function Parameters: 
0:10      'c' (out atomic_uint)
0:12    Sequence
0:12      Branch: Return with expression
0:12        'counter' (layout(binding=0 ) uniform atomic_uint)
0:13      Branch: Return with expression
0:13        Function Call: atomicCounter(au1; (uint)
0:13          'counter' (layout(binding=0 ) uniform atomic_uint)
0:16  Function Definition: main( (void)
0:16    Function Parameters: 
0:?     Sequence
0:19      Sequence
0:19        move second child to first child (uint)
0:19          'val' (uint)
0:19          Function Call: atomicCounter(au1; (uint)
0:19            'counter' (layout(binding=0 ) uniform atomic_uint)
0:20      Function Call: atomicCounterDecrement(au1; (uint)
0:20        'counter' (layout(binding=0 ) uniform atomic_uint)
0:?   Linker Objects
0:?     'counter' (layout(binding=0 ) uniform atomic_uint)


Linked fragment stage:


Shader version: 420
ERROR: node is still EOpNull!
0:5  Function Definition: func(au1; (uint)
0:5    Function Parameters: 
0:5      'c' (in atomic_uint)
0:7    Sequence
0:7      Branch: Return with expression
0:7        Function Call: atomicCounterIncrement(au1; (uint)
0:7          'c' (in atomic_uint)
0:10  Function Definition: func2(au1; (uint)
0:10    Function Parameters: 
0:10      'c' (out atomic_uint)
0:12    Sequence
0:12      Branch: Return with expression
0:12        'counter' (layout(binding=0 ) uniform atomic_uint)
0:13      Branch: Return with expression
0:13        Function Call: atomicCounter(au1; (uint)
0:13          'counter' (layout(binding=0 ) uniform atomic_uint)
0:16  Function Definition: main( (void)
0:16    Function Parameters: 
0:?     Sequence
0:19      Sequence
0:19        move second child to first child (uint)
0:19          'val' (uint)
0:19          Function Call: atomicCounter(au1; (uint)
0:19            'counter' (layout(binding=0 ) uniform atomic_uint)
0:20      Function Call: atomicCounterDecrement(au1; (uint)
0:20        'counter' (layout(binding=0 ) uniform atomic_uint)
0:?   Linker Objects
0:?     'counter' (layout(binding=0 ) uniform atomic_uint)

